frontend/src/static/js/components/webstatus-line-chart-tabbed-panel.ts

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {TemplateResult, html} from 'lit';
+import {
+  AdditionalSeriesConfig,
+  FetchFunctionConfig,
+  WebstatusLineChartPanel,
+} from './webstatus-line-chart-panel.js';
+import {state} from 'lit/decorators.js';
+import {BrowsersParameter} from '../api/client.js';
+import {WebStatusDataObj} from './webstatus-gchart.js';
+
+/**
+ * Extension of the WebstatusLineChartTabbedPanel abstract base class to add
+ * tabbed functionality for multiple views.
+ */
+export abstract class WebstatusLineChartTabbedPanel extends WebstatusLineChartPanel {
+  /**
+   * The processed data objects for each view of the chart, structured for `webstatus-gchart`.
+   * @state
+   * @type {Array<WebStatusDataObj | undefined>}
+   */
+  @state()
+  dataByView?: Array<WebStatusDataObj>;
+
+  /**
+   * Names of the tabs to be displayed in the panel.
+   * @state
+   * @type {Array<string>}
+   */
+  @state()
+  tabViews: Array<string> = [];
+
+  /**
+   * Indicates which tabbed view is active.
+   * @state
+   * @type {number}
+   */
+  @state()
+  currentView: number = 0;
```

Comment:
Where do we modify this property? I may have missed something. I see it being used but I don't see it being changed.

Reply:
This was necessary before the tab-panel implementation, so I've removed it now. 🙂